src/lib.rs

```diff
@@ -1152,7 +1166,9 @@ impl ContextChange {
         match tpsc {
             TypeSchema::Monotype(tp) => self.reify_type(tp),
             TypeSchema::Polytype { variable, body } => {
-                *variable += self.delta;
+                if !self.sacreds.contains(variable) {
+                    *variable += self.delta;
+                }
```

Comment:
The quantified variables don't need to be changed, because they get instantiated into new variables.